upgrader.go

```diff
@@ -170,6 +175,10 @@ func (u *Upgrader) HasParent() bool {
 
 // Upgrade triggers an upgrade.
 func (u *Upgrader) Upgrade() error {
+	if !isSupportedOS() {
```

Comment:
Is this required if `isSupportedOS()` is checked in New?

Reply:
Good point. Removing.